File: 백준/Bronze/1259. 팰린드롬수/팰린드롬수.cc
#include <iostream>
#include <string>
using namespace std;
bool sol(string tmp){
    int i = 0; int j = tmp.length()-1;
    while(i < j){
        if(tmp[i] != tmp[j]) return false;
        i ++; j--;
    }
    return true;
}
int main(){
    while(1){
        string tmp;
        cin >> tmp;
        if (tmp == "0") break;
        if(sol(tmp)) cout<< "yes\n";
        else cout <<"no\n";
    }
}

File: 백준/Bronze/1259. 팰린드롬수/팰린드롬수.py
def sol(str_input):
    i = 0
    j = len(str_input) - 1
    while (i < j) :
        if str_input[i] != str_input[j]:
            return False
        i+=1
        j-=1
    return True




if __name__ == "__main__":
    while (1):
        str_input = input()
        if str_input == "0":
            break
        if sol(str_input):
            print("yes")
        else:
            print("no")